### PR TITLE
Language Server: add go language server implementation

### PR DESCRIPTION
**Pull request by bketelsen**
Howdy!  Stumbled across a blog post describing this and got pretty excited.  I've added Go language server support by copying the python pattern.  It works nicely, from what I can tell.

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by svenefftinge** [comment]
Awesome! Could you please sign-off your work (see https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md#sign-your-work)?

File: src/languages/go/node/go-contribution.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable } from "inversify";
+import { LanguageContribution, IConnection, createServerProcess, forward } from "../../node";
+
+export type ConfigurationType = 'config_win' | 'config_mac' | 'config_linux';
```
Comment (akosyakov) [inline src/languages/go/node/go-contribution.ts]: please remove configuration stuff, it is used only by java ls

**Comment by bketelsen** [comment]
I think I got it.  PTAL.

**Comment by akosyakov** [comment]
@bketelsen Your first commit is still not signed off.
You can sign your last commit by `git commit --amend -s`.
To sign more commits: `git filter-branch --msg-filter "cat - && echo && echo 'Signed-off-by: Your Name <Your.Name@example.com>'" HEAD~2..HEAD` - to sign 2 commits for example
After that you do `git push -f origin HEAD`

**Comment by akosyakov** [comment]
Could you rebase the PR, squash all commits to one with `git rebase -i` and make sure that it is signed off?

**Comment by hexa00** [comment]
Given that the contributor has explicitly agreed to the certificate of origin and is now not responding I propose we take over this PR ?

**Comment by bketelsen** [comment]
You're welcome to land this!  Thanks

**Comment by MarkZ3** [comment]
See https://github.com/theia-ide/theia/pull/432

**Comment by akosyakov** [comment]
@bketelsen thanks again, it will be merged with #432

**Pull request closed without merging**